benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -221,281 +245,115 @@ private synchronized void handleTestRunningState(String testID) {
+            AbortRunningTestTask abortRunningTestTask = new AbortRunningTestTask(testID);
+            Future abortRunningTestTaskFuture = taskExecutorService.submit(abortRunningTestTask);
+
+            try {
+              abortRunningTestTaskFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
+              // nothing to do
+            }
+
+          }
+
+          // once no running tasks we handle the TERMINATED state as usual
+          handleTestState(testID);
           break;
 
-        case GOAL_REACHABLE_NO_REGRESSION_EXPERIMENTS_REMAINING:
-          testModelDAO.setTestRunningState(testID, DETERMINE_EXECUTE_EXPERIMENTS);
+        case TERMINATED:
+          // already terminated
           break;
 
         default:
           // no default
           break;
       }
 
-      // run the next state
-      handleTestState(testID);
 
-    } catch (InterruptedException | ExecutionException e) {
-      // TODO - handle  properly
-      e.printStackTrace();
     } catch (BenchFlowTestIDDoesNotExistException e) {
-      // should not happen since it was added earlier
-      logger.error("test ID does not exist - should not happen");
-    }
-  }
-
-  private synchronized void handleExperimentResult(String testID) {
-
-    logger.info("handleExperimentResult with testID: " + testID);
-
-    HandleExperimentResultTask task = new HandleExperimentResultTask(testID);
-
-    // TODO - should go into a stateless queue (so that we can recover)
-    Future<?> future = taskExecutorService.submit(task);
-
-    // replace with new task
-    testTasks.put(testID, future);
-
-    try {
-
-      // wait for task to complete
-      future.get();
-
-      Boolean hasRegressionModel = explorationModelDAO.hasRegressionModel(testID);
-
-      TestRunningState nextState =
-          hasRegressionModel ? DERIVE_PREDICTION_FUNCTION : VALIDATE_TERMINATION_CRITERIA;
-
-      testModelDAO.setTestRunningState(testID, nextState);
-
-      handleTestState(testID);
-
-    } catch (InterruptedException | ExecutionException e) {
-      // TODO - handle  properly
+      // TODO - handle me
       e.printStackTrace();
-    } catch (BenchFlowTestIDDoesNotExistException e) {
-      // should not happen since it was added earlier
-      logger.error("test ID does not exist - should not happen");
     }
+
   }
 
-  private void waitForRunningTaskToComplete(String testID, Future future,
-      TestRunningState nextState) {
+  public boolean isTerminated(String testID) {
 
     try {
+      BenchFlowTestState benchFlowTestState = testModelDAO.getTestState(testID);
 
-      future.get();
-      testModelDAO.setTestRunningState(testID, nextState);
-      handleTestState(testID);
+      return benchFlowTestState == TERMINATED;
 
-    } catch (InterruptedException | ExecutionException e) {
-      // TODO - handle  properly
-      e.printStackTrace();
     } catch (BenchFlowTestIDDoesNotExistException e) {
```

Comment:
I meant a log message in the `catch`, other than the ` e.printStackTrace();`, but it is fine like it is.